mne/time_frequency/tfr.py

```diff
@@ -2093,15 +2093,55 @@ def __abs__(self):
         epochs.data = np.abs(self.data)
         return epochs
 
-    def average(self):
+    def average(self, method='mean'):
         """Average the data across epochs.
 
+        Parameters
+        ----------
+        method : str | callable
+            How to combine the data. If "mean"/"median", the mean/median
+            are returned. Otherwise, must be a callable which, when passed
+            an array of shape (n_epochs, n_channels, n_freqs, n_time)
+            returns an array of shape (n_channels, n_freqs, n_time).
+            Note that due to file type limitations, the kind for all
+            these will be "average".
+
         Returns
         -------
         ave : instance of AverageTFR
             The averaged data.
+
+        Notes
+        -----
+        Passing in ``np.median`` is considered unsafe when there is complex
+        data because NumPy doesn't compute the marginal median. Numpy currently
+        sorts the complex values by real part and return whatever value is
+        computed. Use with caution. We use the marginal median in the
+        complex case (i.e. the median of each component separately) if
+        one passes in ``median``. See a discussion in scipy:
+
+        https://github.com/scipy/scipy/pull/12676#issuecomment-783370228
+
+        The ``method`` parameter allows e.g. robust averaging.
+        For example, one could do:
+
+            >>> from scipy.stats import trim_mean  # doctest:+SKIP
+            >>> trim = lambda x: trim_mean(x, 0.1, axis=0)  # doctest:+SKIP
+            >>> epochs_tfr.average(method=trim)  # doctest:+SKIP
```

Comment:
have you tested trim_mean with complex numbers? it behaves as expected?

Reply:
No good point. I'll take that out for now since it was just a copy/paste of `Epochs`, which is only real data. The main plus here in this PR was to allow one to use `np.median`, for being "robust to outliers"